app/components/UI/Transactions/index.js

```diff
@@ -397,13 +433,58 @@ class Transactions extends PureComponent {
 		/>
 	);
 
-	render = () => {
-		if (!this.state.ready || this.props.loading) {
-			return this.renderLoader();
+	toggleRetry = () => {
+		this.setState((state) => ({ retryIsOpen: !state.retryIsOpen, errorMsg: undefined }));
+	};
+
+	retry = () => {
+		this.setState((state) => ({ retryIsOpen: !state.retryIsOpen, errorMsg: undefined }));
+		//TODO need a better solution
+		setTimeout(() => {
+			this.onSpeedUpAction(true, this.existingGas, this.existingTx);
+		}, 500);
```

Comment:
```suggestion
	retry = () => {
		this.setState((state) => ({ retryIsOpen: !state.retryIsOpen, errorMsg: undefined }), () => {
		    InteractionManager.runAfterInteractions(() => {
		        this.onSpeedUpAction(true, this.existingGas, this.existingTx);
		    });
		});
```
I _think_ something like this could do the trick?

Reply:
Good suggestion. Thanks